test/Microsoft.ML.Tests/AnomalyDetectionTests.cs

```diff
@@ -29,33 +22,57 @@ public AnomalyDetectionTests(ITestOutputHelper output) : base(output)
         [Fact]
         public void RandomizedPcaTrainerBaselineTest()
         {
-            var mlContext = new MLContext(seed: 1, conc: 1);
-            string featureColumn = "NumericFeatures";
-
             var reader = new TextLoader(Env, new TextLoader.Options()
```

Comment:
>TextLoader [](start = 29, length = 10)

We should update the tests to use the public API if possible off the MLContext object (see the way ML.AnomalyDetection.Trainers... is used)

Reply:
@glebuk , isn't `ML.AnomalyDetection.Trainers.RandomizedPca()` part of the public API?

I can see that it is from here:

https://github.com/dotnet/machinelearning/blob/4acf5aa97ec4e55b32bf612e551e218b5af2a73a/src/Microsoft.ML.Data/MLContext.cs#L41-L44

Reply:
Yes, the comment was about "new TextLoader(Env,,.." which was not.

---
In reply to: [259996353](https://github.com/dotnet/machinelearning/pull/2692#discussion_r259996353) [](ancestors = 259996353)

Reply:
:thumbsup:, reader is derived from mlContext now